pkg/module/resources.go

```diff
@@ -153,60 +153,67 @@ func (rd ResourceDescriptor) String() string {
 	return string(y)
 }
 
-type Definition struct {
-	Resources []Layer
-	Repo      string
-	DefaultCR []byte
-}
-
-// Inspect analyzes the contents of a module and creates resource definitions for each separate layer the module should be split into
+// Inspect analyzes the contents of a module and updates the module definition provided as parameter with all information contained in the module (layers, metadata and resources).
 // Inspect supports:
 // Kubebuilder projects: if a PROJECT file is found and correctly parsed, the project will automatically be generated and layered.
 // Custom module: If not a kubebuilder project, the user has complete freedom to layer the contents as desired via customDefs. Any contents of path not included in the customDefs will be added to the base layer
-func Inspect(path string, cfg *ComponentConfig, customDefs []string, s step.Step, log *zap.SugaredLogger) (*Definition, error) {
-	log.Debugf("Inspecting module contents at [%s]:", path)
-	defs := []Layer{}
+func Inspect(def *Definition, customDefs []string, s step.Step, log *zap.SugaredLogger) error {
```

Comment:
This function is over 100 lines long and supports two main branches of logic. Perhaps splitting this it's implementation into two smaller, unexported functions is possible?

Reply:
Split the function to 2 distinct functions depending on the branch.